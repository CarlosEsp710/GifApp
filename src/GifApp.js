import React, { useState } from "react";

import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifApp = ({ defaultCategory = [] }) => {
  const [categories, setCategories] = useState(defaultCategory);

  return (
    <>
      <h2>Gif App</h2>
      <AddCategory setCategory={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
