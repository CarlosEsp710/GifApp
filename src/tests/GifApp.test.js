import React from "react";
import { shallow } from "enzyme";

import { GifApp } from "../GifApp";

describe("GifApp", () => {
  test("should return component <GifApp/>", () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should return categories list", () => {
    const categories = ["Hannibal", "Iron Man"];
    const wrapper = shallow(<GifApp defaultCategory={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
