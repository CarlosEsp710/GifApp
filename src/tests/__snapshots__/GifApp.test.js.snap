// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifApp should return categories list 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hannibal"
      key="Hannibal"
    />
    <GifGrid
      category="Iron Man"
      key="Iron Man"
    />
  </ol>
</Fragment>
`;

exports[`GifApp should return component <GifApp/> 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
